core/domain/user_jobs_one_off.py

```diff
@@ -470,3 +471,142 @@ def map(user_settings_model):
     def reduce(key, values):
         """Implements the reduce function for this job."""
         yield (key, len(values))
+
+
+class CleanUpUserSubscribersModelOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """Job that cleans up UserSubscribersModel by removing user id if it is
+    present in subscriber ids.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [user_models.UserSubscribersModel]
+
+    @staticmethod
+    def map(item):
+        if item.deleted:
+            return
+        if item.id in item.subscriber_ids:
+            item.subscriber_ids.remove(item.id)
+            item.put()
+            yield ('Removed user from their own subscribers list', item.id)
+
+    @staticmethod
+    def reduce(key, values):
+        values.sort()
+        yield (key, values)
+
+
+class CleanUpCollectionProgressModelOneOffJob(
+        jobs.BaseMapReduceOneOffJobManager):
+    """Job that cleans up CollectionProgressModel by:
+    1. Removing exploration ids which are not a part of the collection.
+    2. Creating CompletedActivitiesModel for completed explorations if it is
+    missing.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [user_models.CollectionProgressModel]
+
+    @staticmethod
+    def map(item):
+        if item.deleted:
+            return
+
+        completed_activities_model = (
+            user_models.CompletedActivitiesModel.get_by_id(item.user_id))
+
+        if completed_activities_model is None:
+            completed_activities_model = (
+                user_models.CompletedActivitiesModel(id=item.user_id))
+            completed_activities_model.exploration_ids = (
+                item.completed_explorations)
+            completed_activities_model.put()
+            yield ('Regenerated Missing CompletedActivitiesModel', item.id)
+
+        if completed_activities_model.deleted:
```

Comment:
Does this case actually occur in prod?

If so, why caused it to be deleted in the first place? I can understand oversights leading to the model not being created, but having it get deleted seems like a more serious problem.

Reply:
No, there is no deleted case, I just added it as a safety net in all the jobs.